Add infrastructure lifecycle and availability sections (#91)
sa-tre/satre-specification: docs/source/standard.md

#### 2.4.1 Deployment management

_The ability of the TRE organisation to instantiate, deploy, change or remove deployed infrastructure._

```{list-table}
:header-rows: 1
:name: tab-deployment-management
* - Statement
  - Guidance
  - Mandatory status
* - You must have a documented procedure for deploying infrastructure.
  - This might, for instance, be a handbook that is followed or a set of automated scripts.
  - Mandatory
* - Where possible, you should automate any repeatable aspects of your deployment.
  - This might involve using infrastructure-as-code tools or simply a series of scripts.
  - Recommended
* - You must test changes before they are used in production.
  - This might involve a separate development environment or another system for testing.
  - Mandatory
* - You could test changes in a development environment that mirrors your production system.
  - Consider the costs and practicality of whether this will work for your situation.
  - Optional
* - You must have a documented procedure for making changes to deployed infrastructure.
  - This refers both to changes that might be expected in the course of normal operation and emergency changes that might be needed.

>   - Mandatory

Reviewer: Maybe it makes more sense to put this above "You must test changes" and "You could have a dev environment"?

Reviewer 2: Also (in future) cross-ref with iso27001/accreditation/change management?

Author: Done

* - You must have a documented procedure for removing infrastructure when it is no longer needed

>   -

Reviewer: ```suggestion
  - Removing unused infrastructure not only reduces costs and management burden but also reduces the attack surface of a TRE and reduces the risk of unaddressed vulnerabilities.
```

I suppose there is also something here like "Don't deploy anything that you don't need, reduce complexity". But that might be more of a design philosophy?

  - Mandatory
```

#### 2.4.2 Capacity management

_The ability of the TRE organisation to ensure the right amount of resources are available at the right time to provide a service._

```{list-table}
:header-rows: 1
:name: tab-capacity-management
* - You must ensure that all projects understand what resources are available and what the associated costs will be before the project starts.
  - For on-premises systems this might be related to the available hardware, for cloud-based systems there might be limits on how many instances of a particular resource (_e.g._ GPUs) can be used. Projects should use this information to understand whether the available resources will be sufficient for their requirements.
  - Mandatory
* - You should ensure that the anticipated needs of projects can be satisfied using available resources.
  - Note that this does not require you to accept requests for additional resources, but rather that promises made about resource availability before a project starts should be honoured wherever possible.
  - Recommended

> * - You must ensure that sufficient budget is available to support the anticipated need for resources.

Reviewer 2: Is this referring to the TRE's budget, or the project's budget in a pay-as-you-go scenario?

  - For cloud-based TREs this may involve budgeting and/or restricting resource consumption on a project-by-project basis.
    For on-premises TREs this may involve managing expectations to match the available resource.
  - Mandatory
* - You must have a procedure for increasing/decreasing available resources.

>   - For cloud-base TREs this may involve scaling resource attributes or deploying additional resources.

Reviewer 2: `scaling resource attributes`: is this the same as scaling compute resources, or does it mean something else?

Author: They might be the same. I was thinking (in a cloud scenario) that you might

- increase the size/power of a VM
- increase the storage capacity of a disk
- increase the processing power of a database

Are these all scaling compute resources?

>   - For on-premises TREs this may involve a procurement process to ensure that necessary resources are available.

Reviewer 2: This will appear in the third column (mandatory/recommended). Were you intending to combine it with the previous line?

* - You must have a procedure to decide when to change capacity.
  - Not all requests for capacity increase must necessarily be granted, but having a clear process will help projects understand when/why/how they can make use of additional capacity.
  - Mandatory
```

#### 2.4.3 Configuration management

_The ability of the TRE organisation to identify, maintain, and verify information on IT assets and configurations in the TRE organisation._

```{list-table}
:header-rows: 1
:name: tab-configuration-management
* - You must have a documented procedure for configuring infrastructure.
  - This might, for instance, be a handbook that is followed or a set of automated scripts.
  - Mandatory
* - You should use configuration management tools to automate application of your configuration wherever possible.

>   - This might involve configuration-as-code tools such as Ansible, Chef, Puppet or Windows Desired State Configuration or simply automated scripts.

Reviewer: ```suggestion
  - This would involve configuration-as-code tools such as Ansible, Chef, Puppet or Windows Desired State Configuration.
```

I would argue a script, on its own, is not configuration management.

Reviewer 2: I think it could be if written correctly, and if version controlled. Doesn't matter though at this stage.

Author: I suggest leaving this as-is for the moment.
